cou/steps/plan.py

```diff
@@ -55,19 +56,21 @@
 logger = logging.getLogger(__name__)
 
 
-async def generate_plan(analysis_result: Analysis, args: CLIargs) -> UpgradePlan:
+async def generate_plan(analysis_result: Analysis, args: CLIargs) -> tuple[UpgradePlan, list[str]]:
     """Generate plan for upgrade.
 
     :param analysis_result: Analysis result.
     :type analysis_result: Analysis
     :param args: CLI arguments
     :type args: CLIargs
-    :return: Plan with all upgrade steps necessary based on the Analysis.
-    :rtype: UpgradePlan
+    :return: Tuple containing a plan with all upgrade steps necessary based on the Analysis,
+        and a list of recorded warning messages from plan generation.
+    :rtype: tuple[UpgradePlan, list[str]]
     """
     _pre_plan_sanity_checks(args, analysis_result)
     target = _determine_upgrade_target(analysis_result)
 
+    warning_messages = []
```

Comment:
Can we create a simple class that inherit from list just to wrap and have a more meaningful object ? E.g:

```python
class PlanWarning(list):
    ....
```
It feels more natural to me to have then a return type `tuple[UpgradePlan, PlanWarning]`